### DISCUSSION TITLE
Logging states and contexts of entire system on any change in v5

### DISCUSSION THREAD

**Discussion by lewisl9029** [statelyai/xstate#3975] [Q&A] [answered]
Hi there! I've been building a [VSCode extension](https://marketplace.visualstudio.com/items?itemName=reflame.agent) using XState, and in v4, I've been able to subscribe to changes of all invoked/spawned actors from the top level service and log them for observability, by spawning/invoking every actor with the `sync: true` option. Without this option, the callback for the top level service won't be called when any spawned/invoked actors are updated.

However, it looks like in v5 this option has been removed. So I'm curious if there's a different approach the team would recommend for keeping tabs on the state of the entire system for observability?

Cheers!

**Comment by Andarist** [comment]
> So I'm curious if there's a different approach the team would recommend for keeping tabs on the state of the entire system for observability?

not yet but there will be

**Reply by lewisl9029** [reply]
Thats awesome to hear, can't wait to try it out! :)

**Comment by davidkpiano** [comment, marked as answer]
The #3969 PR would probably address this, by allowing you to add a custom system-level inspector that observes all actors without having to add `sync: true` (which is no longer in v5).

**Reply by lewisl9029** [reply]
That looks amazing! Sounds like a really cool way to observe the entire system without introducing a ton of chatty sync events that I had to handle and work around before. :)
